src/communication_interface.cc

```diff
@@ -437,3 +482,14 @@ void CommunicationInterface::HandlePause(
     pause_data_.paused_ = true;
   }
 }
+
+// Publish to pause depending on the User Stop status
+void CommunicationInterface::HandleUserStop(
+    const ::lcm::ReceiveBuffer*, const std::string&,
+    const robot_msgs::bool_t* user_stop_msg) {
+
+  PublishPauseToChannel(user_stop_msg->utime,
+                        params_.lcm_stop_channel,
+                        user_stop_msg->data,
```

Comment:
Why do you need to define a new handler for the user stop status? The driver already knows if the robot is user stopped (it is the one publishing to the u-stop channel in the first place